src/app/infra/tokens_repository.ts

```diff
@@ -0,0 +1,33 @@
+import { AccountWalletWithPrivateKey, AztecAddress } from '@aztec/aztec.js';
+import { TokenContract } from '@aztec/noir-contracts/types';
+import { Token, TokensAggregate } from '../model/token_aggregate.js';
+import { TOKEN_LIST } from '../model/token_info.js';
+
+export class TokensRepository {
+  static getTokensAggregate(): TokensAggregate {
+    const iter = TOKEN_LIST.map<Token>(info => {
+      return { info };
+    });
+    return new Array(...iter);
+  }
+
+  static async fetchContracts(tokens: TokensAggregate, wallet: AccountWalletWithPrivateKey) {
```

Comment:
Maybe add return type of method for clarification. `fetchContracts` sounds like method that returns collection of contracts but in this case it returns `TokensAggregate`